src/rules.ts

```diff
@@ -0,0 +1,382 @@
+import {
+  Question,
+  TotalInformationState,
+  InformationState,
+  Move,
+  DomainRelation,
+} from "./types";
+
+type Rules = {
+  [index: string]: (context: TotalInformationState) => {
+    preconditions: boolean;
+    effects: InformationState;
+  };
+};
+
+function relevant(x: DomainRelation): boolean {
+  return ["relevant", "resolves"].includes(x.type);
+}
+function resolves(x: DomainRelation): boolean {
+  return "resolves" === x.type;
+}
+
+const plans = {
```

Comment:
Better move it to the parent machine's context (`isu.ts` or tests), which is provided to DME as input.